devp2p/kademlia.py

```diff
@@ -52,6 +53,12 @@ def __init__(self, pubkey):
         else:
             assert k_id_size == 256
             self.id = big_endian_to_int(sha3(pubkey))
+        self.ping_recv = False
+        self.pong_recv = False
+
+    @property
+    def bonded(self):
+        return self.ping_recv and self.pong_recv
```

Comment:
informal spec from geth:
`bonding` state is valid ~indefinitely in protocol version 4, however: when a node is marked for replacement (new candidate for full bucket should be added) and the known, "bonded" node does not respond to the keep alive ping, it will be 
- replaced by the new candidate BUT
- kept in the "known" state (i.e. bonded, will receive answers to queries) for up to 24h (outside the table)